processing/src/main/java/org/apache/druid/segment/column/ColumnConfig.java

```diff
@@ -22,4 +22,70 @@
 public interface ColumnConfig
 {
   int columnCacheSizeBytes();
+
+  /**
+   * If the total number of rows in a column multiplied by this value is smaller than the total number of bitmap
+   * index operations required to perform to use a {@link LexicographicalRangeIndex} or {@link NumericRangeIndex},
+   * then for any {@link ColumnIndexSupplier} which chooses to participate in this config it will skip computing the
+   * index, in favor of doing a full scan and using a {@link org.apache.druid.query.filter.ValueMatcher} instead.
+   * This is indicated returning null from {@link ColumnIndexSupplier#as(Class)} even though it would have otherwise
+   * been able to create a {@link BitmapColumnIndex}. For range indexes on columns where every value has an index, the
```

Comment:
This was already true of `as` itself, returning `null` has always been an indicator that the index we asked for is not available. I just made it a bit clearer in the docs and also leaned into allowing the things `as` returns to also return null to indicate that an index is not available.

For the types of short circuits done in this PR, i feel like the index supplier has better information to make the decision than anything else above it since it has direct access to the size of dictionary, number of rows to be scanned, etc, but I will think on it